lib/watir/locators/element/selector_builder/xpath.rb

```diff
@@ -13,6 +13,9 @@ class XPath
           def build(selector)
             @selector = selector
 
+            @valid_attributes = Watir::HTMLElement.attribute_list
+            @valid_attributes += Watir.tag_to_class[@selector[:tag_name].to_sym]&.attribute_list if @selector[:tag_name].is_a?(String)
+
```

Comment:
(`+=` results in a bunch of duplicates, and sometimes `@valid_attributes` is nil with code as is`)

Reply:
My assumption is that we want it to work for all attributes defined on an element, meaning attributes defined on the element and all its parent (`HTMLElement` in our case). Hence, the `+=`. There should be no duplicates at all (at least according to `html_elements.rb`) but I can use set instead of array if you want.

Thanks for tag name usign Regexp case, I've totally forgot about it.

Reply:
The attribute list is inherited. I verified with a debug point.

You had Regexp effectively covered when checking for String, I just went the other way (not Regexp)

Reply:
You are right. I've pushed a new version, please check it out.